src/VirtoCommerce.PricingModule.Web/Module.cs

```diff
@@ -144,7 +143,11 @@ public void PostInitialize(IApplicationBuilder appBuilder)
             var inProcessBus = appBuilder.ApplicationServices.GetService<IHandlerRegistrar>();
             inProcessBus.RegisterHandler<PriceChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<LogChangesChangedEventHandler>().Handle(message));
             inProcessBus.RegisterHandler<ProductChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<DeletePricesProductChangedEventHandler>().Handle(message));
-            inProcessBus.RegisterHandler<PriceChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<IndexPricesProductChangedEventHandler>().Handle(message));
+
+            if (settingsManager.GetValue("Pricing.Search.EventBasedIndexation.Enable", false))
```

Comment:
I think there is need to use `ModuleConstants.Settings.General.EventBasedIndexation.Name` instead of direct string.